setoff_workflows/setoff_workflows.py

```diff
@@ -0,0 +1,2622 @@
+                SWConfig.UPLOAD_ARGS["token"] % self.dnanexus_auth,
+            ],
+        )
+
+    def create_upload_multiqc_cmd(self) -> str:
+        """
+        Build dx run command to run upload_multiqc app for the run. This uploads the
+        MultiQC data to the genomics server. The input to the upload_multiqc app is the
+        html_report output of the multiqc app in the format jobid:output_name
+            :return (str): Dx run command for upload_multiqc app
+        """
+        self.rf_obj.rf_loggers.sw.info(
+            self.rf_obj.rf_loggers.sw.log_msgs["building_cmd"],
+            "upload multiqc",
+            self.rf_obj.runfolder_name,
+        )
+        return " ".join(
+            [
+                f'{SWConfig.DX_CMDS["upload_multiqc"]}Upload_MultiQC',
+                SWConfig.APP_INPUTS["upload_multiqc"]["multiqc_html"],
+                SWConfig.APP_INPUTS["upload_multiqc"]["lane_metrics"],
+                SWConfig.APP_INPUTS["upload_multiqc"]["multiqc_output"],
+                SWConfig.UPLOAD_ARGS["depends"],
+                SWConfig.UPLOAD_ARGS["dest"],
+                SWConfig.UPLOAD_ARGS["token"] % self.dnanexus_auth,
+            ]
+        )
+
+    def return_sample_workflow_cmds(self) -> list:
+        """
+        Return sample-level commands. This includes the sample workflow command,
+        and Congenica input and upload commands if required for the sample type
+            :return dx_cmds (list):                         Per-sample workflow commands
+            :return decision_support_upload_cmds (list):    Per-sample decision support upload commands
+        """
+        dx_cmds = [
+            f"PROJECT_ID={self.nexus_project_id}",
+            f"PROJECT_NAME={self.samples_obj.nexus_paths['proj_name']}",
+            f"RUNFOLDER_NAME={self.rf_obj.runfolder_name}",
+            SWConfig.EMPTY_DEPENDS,
+        ]
+        decision_support_upload_cmds = dx_cmds[:]
+
+        if self.samples_obj.pipeline == "pipe":
+            dx_cmds.extend(SWConfig.EMPTY_CP_DEPENDS)
+        for sample_name in self.samples_obj.samples_dict.keys():
+            self.rf_obj.rf_loggers.sw.info(
+                self.rf_obj.rf_loggers.sw.log_msgs["sample_identified"],
+                self.samples_obj.pipeline,
+                sample_name,
+            )
+            dx_cmds.append(
+                self.samples_obj.samples_dict[sample_name]["sample_pipeline_cmd"]
+            )
+            dx_cmds.append(SWConfig.UPLOAD_ARGS["depends_list"])
+
+            if self.samples_obj.pipeline == "pipe":
+                # Add to gatk depends list because RPKM must depend only upon the
+                # sample workflows completing successfully, whilst other downstream
+                # apps depend on all prior jobs completing succesfully
+                dx_cmds.append(SWConfig.UPLOAD_ARGS["depends_list_gatk"])
+
+            if self.samples_obj.pipeline in ["wes", "pipe"]:
```

Comment:
what about TSO decision support commands?
I suspect these might be written somewhere else and I've missed/forgotten as this review is happening over a long period of time. If this is the case, please can a comment be added here (or somewhere appropriate nearby) to confirm this.

Reply:
Yeah these are created elsewhere - Done